### ISSUE TITLE
Add foreign keys... generate ERD

### ISSUE THREAD

**Issue by ross-spencer** [exponential-decay/sqlitefid#9]
The foreign keys are not set on the database. This doesn't seem to be a big problem so far. 

A quick and dirty approach is as follows: 
```diff
diff --git a/libs/GenerateBaselineDBClass.py b/libs/GenerateBaselineDBClass.py
index 6b47790..0a7b322 100644
--- a/libs/GenerateBaselineDBClass.py
+++ b/libs/GenerateBaselineDBClass.py
@@ -178,6 +178,7 @@ class GenerateBaselineDB:
                 table = self.createfield(table, column, "integer")
             else:
                 table = self.createfield(table, column)
+        table = "{} FOREIGN KEY(FILE_ID) REFERENCES IDRESULTS(FILE_ID), ".format(table)
         table = table.rstrip(", ") + ")"
         self.execute_create(table)
 
@@ -192,6 +193,8 @@ class GenerateBaselineDB:
                 table = self.createfield(table, column, "boolean")
             else:
                 table = self.createfield(table, column)
+        table = "{} FOREIGN KEY(ID_ID) REFERENCES IDRESULTS(ID_ID), ".format(table)
+        table = "{} FOREIGN KEY(NS_ID) REFERENCES NSDATA(NS_ID), ".format(table)
         table = table.rstrip(", ") + ")"
         self.execute_create(table)
 
@@ -219,6 +222,7 @@ class GenerateBaselineDB:
     def create_indices(self):
         self.execute_create("CREATE INDEX HASH ON {} (HASH)".format(self.FILEDATATABLE))
         self.execute_create("CREATE INDEX NAME ON {} (NAME)".format(self.FILEDATATABLE))
+        self.execute_create("CREATE INDEX FILE_PATH_HASH ON {} (FILE_PATH, HASH)".format(self.FILEDATATABLE))
         self.execute_create("CREATE INDEX PUID ON {} (ID)".format(self.IDTABLE))
 
     def execute_create(self, query):
```
This whole segment of code could do with an overhaul. Being "dynamic" may actually be a negative and affect readability. The tables themselves are, and have remained pretty static. 

Once we have our foreign keys setup we can then create an ERD: 

![image](https://user-images.githubusercontent.com/1880412/154716540-a537b47c-c820-40d1-88ed-cc22413b54e8.png)

Using the command:

```bash
docker run -v "${PWD}/sspy/output:/output" -v "${PWD}/sspy:/sspy" schemaspy/schemaspy:latest -configFile sspy/sprops -t sqlite-xerial -debug
```

And config `sprops`:

```conf
schemaspy.t=sqlite
schemaspy.db=sspy/example.db
schemaspy.sso=true
schemaspy.o=sspy/output/
schemaspy.cat=example
schemaspy.s=example
schemaspy.dp=sspy/sqlite-jdbc-3.30.1.jar
```

**Comment by ross-spencer** [comment]
NB. `self.execute_create("CREATE INDEX FILE_PATH_HASH ON {} (FILE_PATH, HASH)".format(self.FILEDATATABLE))` isn't needed. This is a separate index used in testing.

**Comment by ross-spencer** [comment]
Resolved via 90a978732
